client/tests/integration_test.rs

```diff
@@ -668,18 +717,98 @@ mod sign_deploy {
                 unsigned_file_path.to_str().unwrap(),
                 deploy_params::test_data_valid(),
                 session_params::test_data_with_package_hash(),
-                payment_params::test_data_with_name()
+                payment_params::test_data_with_name(),
+                false
             )
             .map_err(ErrWrapper),
             Ok(())
         );
         assert!(casper_client::sign_deploy_file(
             unsigned_file_path.to_str().unwrap(),
             "<this is not a path>",
-            ""
+            "",
+            false
         )
         .is_err());
     }
+
+    #[test]
+    fn should_fail_with_existing_file() {
+        let temp_dir = TempDir::new()
+            .unwrap_or_else(|err| panic!("Failed to create temp dir with error: {}", err));
+        let unsigned_file_path = temp_dir.path().join("test_deploy.json");
+        let signed_file_path = temp_dir.path().join("signed_test_deploy.json");
+        assert_eq!(
+            casper_client::make_deploy(
+                unsigned_file_path.to_str().unwrap(),
+                deploy_params::test_data_valid(),
+                session_params::test_data_with_package_hash(),
+                payment_params::test_data_with_name(),
+                false
+            )
+            .map_err(ErrWrapper),
+            Ok(())
+        );
+        assert_eq!(
+            casper_client::sign_deploy_file(
+                unsigned_file_path.to_str().unwrap(),
+                "../resources/local/secret_keys/node-1.pem",
+                signed_file_path.to_str().unwrap(),
+                false
+            )
+            .map_err(ErrWrapper),
+            Ok(())
+        );
+        assert_eq!(
+            casper_client::sign_deploy_file(
+                unsigned_file_path.to_str().unwrap(),
+                "../resources/local/secret_keys/node-1.pem",
+                signed_file_path.to_str().unwrap(),
+                false
+            )
+            .map_err(ErrWrapper),
+            Err(Error::FileAlreadyExists(signed_file_path).into())
+        );
```

Comment:
Also here, let's check the failed attempt doesn't modify the pre-existing output file.